GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -1650,6 +1671,14 @@ private void UnstagedFileContext_Opening(object sender, System.ComponentModel.Ca
             openWithToolStripMenuItem.Enabled = !isAnyDeleted;
             deleteFileToolStripMenuItem.Enabled = !isAnyDeleted;
             openContainingFolderToolStripMenuItem.Enabled = !isAnyDeleted;
+
+            UnstagedFileContext.AddUserScripts(runScriptToolStripMenuItem, ExecuteCommandUnstaged, (script) => script.OnEvent == ScriptEvent.ShowInFileList, UICommands);
+
+            bool ExecuteCommandUnstaged(int command)
+            {
+                IScriptsRunner scriptsRunner = UICommands.GetRequiredService<IScriptsRunner>();
+                return scriptsRunner.RunScript(command, owner: this, UICommands, new ScriptOptionsProvider(Unstaged, () => _fullPathResolver, () => SelectedDiff.CurrentFileLine));
+            }
```

Comment:
I dunno, this is really opening up for a race condition on fast clickers or slow computers, where you right click one list at the same time as you hit tab to jump to the other, and it gets confused and runs the command on the wrong files. I think it's better to be safe here.

Reply:
The UI thread processes key presses, mouse clicks and focus changes in order. I do not see how this could open up for a race condition.
Asynchronous tasks - as used for loading the diff - are not related.

Reply:
I'm not entirely sure about that. At the very least, I was able to open a context menu while focus was on a completely different window, by holding the right mouse button, hitting alt+tab, and then letting go of the right mouse button. With more effort, or a slightly more complex process, I might be able to get the context menu to open while the wrong control is focused.

Even if I wasn't able to, I don't think it's worthwhile for the sake of saving 2-3 lines of code. Unless writing real-time software that relies complex use of unfair locks/semaphores to avoid queue allocations, there's no real reason to intentionally write code that only works due to multiple aspects aligning just right. It only makes for difficult-to-maintain code.

See the prior issues regarding possible multiple calls to `RescanChanges` on what such difficult-to-maintain code may look like.

Reply:
I do not want to maintain those redundant lines of code just in order to prevent someone who wants to shoot oneself in the foot, from nonsense clicking.

Reply:
It feels weird to introduce tech debt on purpose, to save 2-3 lines. But it won't be me paying for it, so as you wish.